Source/ACE.Server/Factories/LootGenerationFactory_Melee.cs

```diff
@@ -79,86 +89,62 @@ private static void MutateMeleeWeapon(WorldObject wo, TreasureDeath profile, boo
             int wieldDiff = GetWieldDifficulty(profile.Tier, WieldType.MeleeWeapon);
             WieldRequirement wieldRequirments = WieldRequirement.RawSkill;
 
-            // Weapon Types
-            // 0 = Heavy
-            // 1 = Light
-            // 2 = Finesse
-            // default = Two Handed
-            switch (weaponType)
+            switch (wieldSkillType)
             {
-                case 0:
-                    // Heavy Weapons
-                    wieldSkillType = Skill.HeavyWeapons;
-
-                    switch (subtype)
+                case Skill.HeavyWeapons:
+                    switch (wo.W_WeaponType)
```

Comment:
also hesistant about this change, as it is highly dependent on the PropertyInt.WeaponType fields being set consistently across every item in the database, which would need to be verified. it needs to be ensured that this logic matches up completely with the previous logic